run_tests.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Searches for and runs all tests in this repository."""
+# Testing
+from absl.testing import absltest
+
+# Logging
+from absl import logging
+
+# We use these to import all tests
+import glob
+import subprocess
+import os
+import inspect
+from importlib import import_module
+
+# Folders which can have tests
+TEST_FOLDERS = ['./tests', './src', './evaluations']
+
+
+def find_and_add_tests():
+    """Finds all tests in TEST_FOLDERS directories and adds them to the global
+    namespace for this module.
+
+    Side Effects:
+        Adds tests to global namespace so that absltest.main() can find
+        and run them.
+    """
+    # Get all python files
+    list_of_files = []
+    for folder in TEST_FOLDERS:
+        # Check subdirectories and first level directory for python files
+        list_of_files.extend(glob.glob(folder + '/**/*.py', recursive=True))
+
+    # Get all python files that are in tests/ folders
+    test_files = glob.fnmatch.filter(list_of_files, '*tests/*.py')
+
+    # Turn file names into module names ready for import
+    # (For example: ./tests/test.py --> tests.test)
+    test_modules = map(
+        lambda x: '.'.join(os.path.normpath(x[:-3]).split(os.path.sep)),
```

Comment:
This is a little too complex for a one liner. One option would be to add a detailed comment about what it's doing. Another would be to make a module level function and drop the lambda and add the comment. I'd prefer the later.

Reply:
I agree that this function is too complicated for a one-liner. I don't believe it's complicated nor reusable enough for a module-level function, so I made it a method-level function in b8d83f0

Reply:
Perfect! Thanks.